regression/Makefile

```diff
@@ -13,10 +13,46 @@ DIRS = ansi-c \
        test-script \
        # Empty last line
 
+.DEFAULT_GOAL := test
+
+# How many cores are available
+# Windows not supported = 1
+# CI has an issue with reporting correct number
+# `getconf _NPROCESSORS_ONLN` works on Linux and OS X
+ifeq ($(OS),Windows_NT)
+	NJOB = 1
+else
+  ifdef CI
+  	NJOB = 2
+  else
+    NJOB = $(shell getconf _NPROCESSORS_ONLN)
+  endif
+endif
 
 test:
 	$(foreach var,$(DIRS), $(MAKE) -C $(var) test || exit 1;)
 
+###############################################################################
+# Run tests using GNU parallel 20170722 or above
+# `make -C regression test-parallel`
+#
+# Prereq:
+# - Install GNU parallel:
+#   - ubuntu
+#   	(wget -O - pi.dk/3 || curl pi.dk/3/ || \
+#     fetch -o - http://pi.dk/3) | bash
+#   - osx
+#			brew install parallel
+###############################################################################
+test-parallel:
+	parallel --halt now,fail=1 \
+		--jobs $(NJOB) \
+		--bar \
+		--joblog regression.stats \
+		$(MAKE) -C {} test \
+		::: $(DIRS)
```

Comment:
I don't understand why neither make's built-in parallelisation nor the parallelisation support in test.pl is being used?

Reply:
I would love to use something more platform independent, but I have following problems:
* `make` - output from parallel run of `test.pl` is mixed so it is hard to see what has failed and stats from `test.pl` are useless.
* `test.pl` - for some reason it do not behave deterministic. Some tests fail and then pass on next run. Maybe it is the way how tests are written. Honestly, I do not know.